apt-hook/20apt-esm-hook.conf

```diff
@@ -1,3 +1,7 @@
+APT::Update::Pre-Invoke {
+	"[ ! -e /run/systemd/system ] || systemctl start apt-news.service || true";
+};
```

Comment:
Updated based on @julian-klode's suggestions. The service file is here at [systemd/apt-news.service](https://github.com/canonical/ubuntu-advantage-client/pull/2338/files#diff-6ac1fa2387a109e054cc5986bba22c26e5d561717959597f4058bdd64af48b2c)

Reply:
@basak does this mechanism look okay to you as well?

Reply:
Works with the `Type=exec` service and no timeout in the unit file. Maybe add a timeout to the python script itself, but consider if you want to exit non-zero if the timeout is hit or not, it will affect the systemd service unit state.

Reply:
I remember needing to use --no-block otherwise it waits for one shot units to finish or something, but I don't know how it's solved here.

Reply:
It was changed to `Type=exec`. `oneshot` does indeed block.

Reply:
@panlinux So I noticed that the version of systemd on xenial doesn't recognize `Type=exec`. I think it falls back to "simple", but I decided to change the type back to "oneshot" (which is recognized back on xenial). I think "oneshot" most accurately describes the use of the service.

To avoid blocking on the oneshot service, I've added `--no-block` to the` systemctl start` command. It's working well in my tests.